### PR TITLE
Have functional localhost proxy command, need to work more on deleting

### PR DESCRIPTION
**Pull request by aaron-prindle**
proxies if a service is deleted.  Also still need to add tests.

This is done by adding a Watch to all k8s services and then creating a proxy from the localhost:port to the VM-ip:port.
#38

### CODE REVIEW

File: pkg/minikube/cluster/cluster.go [hunk 1]
```diff
@@ -447,21 +448,25 @@ func getServicePort(namespace, service string) (int, error) {
...
-		return 0, fmt.Errorf("Error getting %s service: %s", service, err)
-	}
+func GetServicePortFromService(svc *kubeApi.Service) (int, error) {
 	nodePort := 0
 	if len(svc.Spec.Ports) > 0 {
 		nodePort = int(svc.Spec.Ports[0].NodePort)
 	}
 	if nodePort == 0 {
-		return 0, fmt.Errorf("Service %s does not have a node port. To have one assigned automatically, the service type must be NodePort or LoadBalancer, but this service is of type %s.", service, svc.Spec.Type)
+		return 0, fmt.Errorf("Service does not have a node port. To have one assigned automatically, the service type must be NodePort or LoadBalancer, but this service is of type %s.", svc.Spec.Type)
```
Comment (jimmidyson) [inline pkg/minikube/cluster/cluster.go]: Prefer to output the service name still, `svc.ObjectMeta.Name` iirc.

**Comment by jimmidyson** [comment]
I'm not 100% sold on the usefulness of proxying the service node port. It's too difficult to discover compared to the actual service port(s), i.e. kubectl doesn't output the node port. The problems with using the actual service port are unable to bind to privileged ports & port conflicts (multiple services can use same port thanks to each service having it's own IP).

How do you expect a user to use this?

**Comment by dlorenc** [comment]
I think this was meant more as a proof of concept. I agree it's minor, but exposing things on localhost is a minor usability win over the VM-specific IP. It's come up a few times in issues, but maybe I misunderstood the use cases:

#38 
#198 

and in the Docker for Mac issue: #99

**Comment by dlorenc** [comment]
This could also be extended for something like #347

**Comment by jimmidyson** [comment]
Yeah agree that perhaps it is a slight UX improvement, although I'm unsure how you'd handle port conflicts (something on host already using specified node port). I have no idea about how we can improve this to act like an internal VPN across OSes - seems possible on OS X via hyperkit, but would prefer a consistent UX across OSes.

**Comment by rubenv** [comment]
Is there any way I can help to get this moving forward?

**Pull request closed without merging**